crates/staged-sync/src/utils/init.rs

```diff
@@ -93,38 +94,32 @@ pub fn insert_genesis_state<DB: Database>(
     tx: &<DB as DatabaseGAT<'_>>::TXMut,
     genesis: &reth_primitives::Genesis,
 ) -> Result<(), InitDatabaseError> {
-    let mut account_cursor = tx.cursor_write::<tables::PlainAccountState>()?;
-    let mut storage_cursor = tx.cursor_write::<tables::PlainStorageState>()?;
-    let mut bytecode_cursor = tx.cursor_write::<tables::Bytecodes>()?;
+    let mut state = PostState::default();
 
-    // Insert account state
     for (address, account) in &genesis.alloc {
         let mut bytecode_hash = None;
-        // insert bytecode hash
         if let Some(code) = &account.code {
-            let hash = keccak256(code.as_ref());
-            bytecode_cursor.upsert(hash, Bytecode::new_raw_with_hash(code.0.clone(), hash))?;
-            bytecode_hash = Some(hash);
+            let bytecode = Bytecode::new_raw(code.0.clone());
+            // FIXME: Can bytecode_hash be Some(Bytes::new()) here?
```

Comment:
no, if they have bytecode the hash must match since that is how we load bytecode from the database. otherwise the account would have no bytecode associated with it

```suggestion

```